plugins/net_plugin/net_plugin.cpp

```diff
@@ -1667,14 +1739,24 @@ namespace eosio {
          sb = cc.fetch_block_by_number( num ); // thread-safe
       } FC_LOG_AND_DROP();
       if( sb ) {
-         enqueue_block( sb, true );
+         if( block_sync_rate_limit > 0 && peer_syncing_from_us ) {
```

Comment:
please add comment in a code of what is happening here. It took me a while to understand your approach so I think having a comment will be helpful

Reply:
Comment added.